Marlin/Marlin_main.cpp

```diff
@@ -4279,6 +4282,144 @@ inline void gcode_G92() {
 
 #endif // ULTIPANEL || EMERGENCY_PARSER
 
+
+
+ 
+ 
+/**
+  * M3 & M4 spindle commands
+  *  S=0 turns off spindle
+  *  if speed pwm output not defined then just turns it on
+  *
+  *  need at least 12.8KHz (50Hz * 256) for spindle PWM 
+  *    must use hardware PWM because all ISRs are too slow 
+  *        Stepper::isr too slow
+  *        temperature ISR  too slow
+  *
+ 
+ NOTE - WGM bits for timers 3, 4 & 5 need to be either 0001 or 0101.
+        Any other setting will result in a PWM signal that doesn't
+        go from 0 volts to 5 volts.
+        
+        The system automatically sets those bits to 0001 so special
+        initialization is not needed.
+        
+        WGM bits for timer 2 are automatically set by the system to
+        001.  This produces an acceptable 0 volts to 5 volts signal.
+        Special initialization is not needed.
+        
+ NOTE - A minimum PWM frequency of 50 Hz is needed.  All of the 
+        pre-scaler factors for timers 2, 3, 4 & 5 are acceptable. 
+ 
+ */
+ 
+  //  SPINDLE_ENABLE_PIN needs an external pullup or else could turn on 
+  //  spindle  during power up or when connecting to the host (usually 
+  //  goes through a reset which sets all I/O pins to tristate)
+ 
+  // PWM duty cycle goes from 0 (off) to 255 (always on) 
+
+#if ENABLED(SPINDLE_ENABLE)
+
+#define SPINDLE_POWER_UP    refresh_cmd_timeout();  \
```

Comment:
Macros like this must be wrapped in `do{ ... }while(0)` and should not end with a semicolon.